modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/raft/MetaStorageWriteHandler.java

```diff
@@ -72,6 +80,29 @@ public class MetaStorageWriteHandler {
     void handleWriteCommand(CommandClosure<WriteCommand> clo) {
         WriteCommand command = clo.command();
 
+        CommandClosure<WriteCommand> resultClosure;
+
+        if (command instanceof IdempotentCommand) {
+            CommandId commandId = ((IdempotentCommand) command).id();
+            IdempotentCommandCache cache = idempotentCommandCache.get(commandId);
```

Comment:
The var naming is confusing, please check another comment for more details.

Reply:
if you mean `cache`, I renamed it

Reply:
Yep, cache.